From 55ec54de63f738bc5cb108fcdaeda76917afc083 Mon Sep 17 00:00:00 2001
From: Steeve Morin <steeve.morin@gmail.com>
Date: Wed, 31 Jan 2024 22:12:58 +0100
Subject: [PATCH] [PJRT C API] Expose //xla/pjrt:mlir_to_hlo to public
 visibility

This is needed because it implements the stablehlo downgrade we need
to be able to lower to PJRT plugins that uses lower stablehlo versions
that the one we use in the main XLA tree.

1. https://github.com/openxla/xla/commit/17d32497875c8a528f7731f6361e5c16c462279d#diff-7f335a78926dafcc06a6940ab5c1d5496f74a3431629e17e54df961e720e7574R230
2. https://groups.google.com/a/openxla.org/g/openxla-discuss/c/hEoA4V5DZF0
3. https://github.com/openxla/stablehlo/pull/1887

Signed-off-by: Steeve Morin <steeve.morin@gmail.com>
---
 xla/pjrt/BUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xla/xla/pjrt/BUILD b/xla/xla/pjrt/BUILD
index dc66fb4cbc..d7b78d59e2 100644
--- a/xla/pjrt/BUILD
+++ b/xla/pjrt/BUILD
@@ -599,7 +599,7 @@ cc_library(
     name = "mlir_to_hlo",
     srcs = ["mlir_to_hlo.cc"],
     hdrs = ["mlir_to_hlo.h"],
-    visibility = [":friends"],
+    visibility = ["//visibility:public"],
     deps = [
         "//xla:debug_options_flags",
         "//xla:util",
